x/ccv/provider/keeper/double_vote.go

```diff
@@ -0,0 +1,107 @@
+package keeper
+
+import (
+	"bytes"
+	"fmt"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+	ccvtypes "github.com/cosmos/interchain-security/v2/x/ccv/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// HandleConsumerDoubleVoting verifies a double voting evidence for a given a consumer chain ID
+// and a public key and, if successful, executes the jailing of the malicious validator.
+func (k Keeper) HandleConsumerDoubleVoting(
+	ctx sdk.Context,
+	evidence *tmtypes.DuplicateVoteEvidence,
+	chainID string,
+	pubkey cryptotypes.PubKey,
+) error {
+	// verifies the double voting evidence using the consumer chain public key
+	if err := k.VerifyDoubleVotingEvidence(ctx, *evidence, chainID, pubkey); err != nil {
+		return err
+	}
+
+	// get the validator's consensus address on the provider
+	providerAddr := k.GetProviderAddrFromConsumerAddr(
+		ctx,
+		chainID,
+		types.NewConsumerConsAddress(sdk.ConsAddress(evidence.VoteA.ValidatorAddress.Bytes())),
+	)
+
+	// execute the jailing
+	k.JailValidator(ctx, providerAddr)
+
+	k.Logger(ctx).Info(
+		"confirmed equivocation",
+		"byzantine validator address", providerAddr.String(),
+	)
+
+	return nil
+}
+
+// VerifyDoubleVotingEvidence verifies a double voting evidence
+// for a given chain id and a validator public key
+func (k Keeper) VerifyDoubleVotingEvidence(
+	ctx sdk.Context,
+	evidence tmtypes.DuplicateVoteEvidence,
+	chainID string,
+	pubkey cryptotypes.PubKey,
+) error {
+	if pubkey == nil {
+		return fmt.Errorf("validator public key cannot be empty")
+	}
+
+	// Note that since we're only jailing validators for double voting on a consumer chain,
+	// the age of the evidence is irrelevant and therefore isn't checked.
+
+	// TODO: check the age of the evidence once we slash
```

Comment:
Remove this TODO (?)
Even the current approach we intend for following on slashing won't check the age.